src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -257,13 +263,15 @@ class RAY_EXPORT PythonGcsClient {
 
  private:
   void PrepareContext(grpc::ClientContext &context, int64_t timeout_ms) {
-    if (timeout_ms != -1) {
-      context.set_deadline(std::chrono::system_clock::now() +
-                           std::chrono::milliseconds(timeout_ms));
+    if (timeout_ms >= 0) {
+      context.set_deadline(
+          std::chrono::system_clock::now() +
+          std::chrono::milliseconds(timeout_ms > 0 ? timeout_ms : GetGcsTimeoutMs()));
     }
     if (!cluster_id_.IsNil()) {
       context.AddMetadata(kClusterIdKey, cluster_id_.Hex());
     }
+    context.set_wait_for_ready(true);
```

Comment:
can we assert failure if timeout_ms < 0, or not enable this? It means it can hang forever if there's no timeout right (e.g., the other process dies, which is pretty common)?